File: backend/app/src/db/models/Item.ts
import {
  AutoIncrement,
  BelongsTo,
  Column,
  CreatedAt,
  DataType,
  ForeignKey,
  HasMany,
  Model,
  PrimaryKey,
  Table,
} from "sequelize-typescript";
import Graphic from "./Graphic";

@Table({ tableName: "Item" })
class Item extends Model<ItemAttributes> {

  @PrimaryKey
  @AutoIncrement
  @Column(DataType.INTEGER)
  declare id: number;

  @Column(DataType.STRING)
  declare name: string;

  @Column(DataType.STRING)
  declare itemId: string;

  @Column(DataType.STRING)
  declare key: string;

  @Column(DataType.STRING)
  declare colorLine: string;

  @Column(DataType.BOOLEAN)
  declare isDicoveryItem: boolean;

  @ForeignKey(() => Graphic)
  @Column(DataType.INTEGER)
  declare graphicId: number;

  @BelongsTo(() => Graphic)
  declare graphic: Graphic;
}

interface ItemAttributes {
  id: number;
  name: string;
  itemId: string;
  key: string;
  colorLine: string;
  graphicId: number;
  isDicoveryItem: boolean;
}

export default Item;

File: backend2/models/item/item.py
from sqlalchemy import ForeignKey
from sqlalchemy.orm import relationship
from db import db

class Item(db.Model):
    """
    Item model representing data points in a graphic.
    """
    __tablename__ = "items"

    # Primary Key
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)

    # Columns
    name = db.Column(db.String, nullable=False)
    item_id = db.Column(db.String, nullable=False)
    key = db.Column(db.String, nullable=False)
    color_line = db.Column(db.String, nullable=False)
    is_discovery_item = db.Column(Boolean, nullable=False)

    # ForeignKey
    graphic_id = db.Column(db.Integer, ForeignKey("graphics.id"), nullable=False)

    # Relationship to Graphic
    graphic = relationship("Graphic", back_populates="items")

    def to_dict(self):
        """
        Convert the model instance to a dictionary.
        """
        return {
            "id": self.id,
            "name": self.name,
            "item_id": self.item_id,
            "key": self.key,
            "color_line": self.color_line,
            "is_discovery_item": self.is_discovery_item,
            "graphic_id": self.graphic_id,
        }